Cargo.toml

```diff
@@ -73,14 +94,14 @@ nu-protocol = { path = "./crates/nu-protocol", version = "0.91.1" }
 nu-std = { path = "./crates/nu-std", version = "0.91.1" }
 nu-utils = { path = "./crates/nu-utils", version = "0.91.1" }
 
-reedline = { version = "0.30.0", features = ["bashisms", "sqlite"] }
+reedline = { workspace = true, features = ["bashisms", "sqlite"] }
 
-crossterm = "0.27"
-ctrlc = "3.4"
-log = "0.4"
-miette = { version = "7.1", features = ["fancy-no-backtrace", "fancy"] }
+crossterm = { workspace = true }
+ctrlc = { workspace = true }
+log = { workspace = true }
+miette = { workspace = true, features = ["fancy-no-backtrace", "fancy"] }
 mimalloc = { version = "0.1.37", default-features = false, optional = true }
-serde_json = "1.0"
+serde_json = { workspace = true }
 simplelog = "0.12"
```

Comment:
Why do we use `workspace` for some dependencies but not others?

Reply:
It's purely because I didn't want to spend a lot of time doing all of them before knowing if this change would be accepted or not 😅

We can either keep adding to this PR, or progressively do the rest in subsequent PRs.

Reply:
Ha, that's a very fair reason. Sounds good to me.

Reply:
(as in I'm OK with subsequent PRs)

Reply:
Also sorry, I just noticed that you explained this in the PR description.